### ISSUE TITLE
Won't retry more than once

### ISSUE THREAD

**Issue by ivoiv** [JustinBeckwith/retry-axios#31]
EDIT: I swapped retry-axios for axios-retry @ https://github.com/softonic/axios-retry , using the same logical flow and it works fine with proper repeats. What's so different about retry-axios' implementation?

Hello.

I've been trying to get this wonderful interceptor all day but I just can't get it to work with my React app.

I have the following code.

```
export default{
  get(url, conf = {}) {
    const client = getClient();
    client.defaults.raxConfig = {
      retry: 5,
      instance: client,
      noResponseRetries: 5,
      statusCodesToRetry: [[100, 199], [400, 429],[500, 599]],
      onRetryAttempt: (err) => {
        const cfg = rax.getConfig(err);
        console.log(`Retry attempt #${cfg.currentRetryAttempt}`);
      }
    }
    const interceptorId = rax.attach(client)
    return client.get(url, conf)
  }
}
```

Basically another apiCall.js file calls the exported function, it creates an axios instance with some basic options via getClient(), attaches the raxConfig, attaches rax itself and then returns the get request.

The apiCall.js file then resolves the promises via .then() and .catch().

The problem im experiencing is no matter what I do retry-axios only retries once.

I know it uses the config because it properly reacts to changes in statusCodesToRetry, but doesn't do anything after the first retry.

I'm really desperate here and any help is appreciated. Thank you.

**Comment by erzzo** [comment]
same here

**Comment by erzzo** [comment]
Okay, my problem was that in the config, httpMethodsToRetry didn't include the POST method by default.

**Comment by pdf13** [comment]
I have the same problem and change to softonic/axios-retry to make it work.
=(

**Comment by ukcrpb6** [comment]
Had a similar issue, this was due to the use of axios defaults which is copied by reference, *not cloned* to each request. This meant the number of current retries stored in the config by the request error handler is global, leading to effectively 1 set of retries.

This can be fixed using a request interceptor to set the config instead such as the following (typescript):

```typescript
import { AxiosInstance, AxiosRequestConfig } from 'axios';
import { attach, RetryConfig } from 'retry-axios';

type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
type PartialRaxConfig = { raxConfig?: RetryConfig } & AxiosRequestConfig;

export function applyAxiosRetry(
  instance: AxiosInstance,
  options?: Omit<RetryConfig, 'instance'>,
): AxiosInstance {
  instance.interceptors.request.use((config: PartialRaxConfig) => {
    config.raxConfig = config.raxConfig || { instance, ...options };
    return config;
  });

  attach(instance);
  return instance;
}
```

usage:

```typescript
import axios from 'axios';
import { applyAxiosRetry } from './axios-retry.utils';

const instance = axios.create({ /* ... */ });
applyAxiosRetry(instance, {
  retry: 5,
  /* ... */
});
```

**Comment by brandoncc** [comment]
I had a similar problem I was getting an error that `httpMethodsToRetry.indexOf` wasn't a function. That was only after it had already retried once. It seems it was clobbering my config somehow. I was setting every option in `instance.defaults.raxConfig`, even if it just got set to the default (to show what options we could change). `axios-retry` works fine for me too. I prefer the options API of this package though so I wish it worked for me.

**Comment by JustinBeckwith** [comment]
@brandoncc I'm pretty sure we addressed the `indexOf` problem in #83, which should be in the latest release :)

**Comment by gvmohzibat** [comment]
@ukcrpb6 Your solution works but then the `currentRetryAttempts` will always be equal to 1 and therefore, it'll retry infinitely. I switched to `axios-retry`.
